### PR TITLE
Feature/delete

### LINKED ISSUE #34: Delete functionality
## Value Proposition
As a tutor, I would like to have the ability to delete tutorials I have created to remove outdated tutorials from distribution.

## Description
capstone-userstory-13

## Acceptance Criteria
- [x] in `user` page, every tutorial list item displays a `delete` icon
- [x] a modal opens when the `delete` icon is clicked
- [x] the modal displays the question `Do you really want to delete this tutorial?`, a `no, abort` and a `yes, delete` button
- [x] the modal closes when one of the buttons or the background is clicked
- [x] additionally, the tutorial will be deleted when the `yes, delete` button was clicked

## Tasks
- [x] create new branch
- [x] add `delete` icon to tutorial list item
- [x] create functionality to remove a specific document from database collection (accepts tutorial ID)
- [x] create `modal` component and make use of `Button` component to insert the two buttons
- [x] add state for visibility of modal
- [x] add functionality to `yes, delete` button

### PR DESCRIPTION
**Pull request by kai-franke**
This PR:
- adds a `modal` component
- adds a DELETE API method
- adds the ability to delete individual tutorial

### CODE REVIEW

File: pages/user.js [hunk 1]
```diff
@@ -36,12 +33,44 @@ export async function getServerSideProps({ req }) {
...
+      {showModal && (
+        <Modal click={() => setShowModal(false)}>
+          <TbTrash style={{ fontSize: "1.5em" }} />
+          <Paragraph>Do you really want to delete this tutorial?</Paragraph>
+          <ButtonContainer>
+            <Button isPrimary>
+              <TbX style={leftIconStyle} /> no, cancel
+            </Button>
+            <Button onClick={deleteTutorial}>
+              yes, delete <TbCheck style={reftIconStyle} />
```
Comment (JannikUekoetter) [inline pages/user.js]: Am Satzanfang bitte Groß- und Kleinschreibung beachten.

File: pages/user.js [hunk 2]
```diff
@@ -50,13 +79,18 @@ export default function TutorialsPage({ userTutorials, userName }) {
         <Message>You haven&apos;t created any tutorials yet</Message>
```
Comment (JannikUekoetter) [inline pages/user.js:79]: Die Darstellung im Satz kommt mir komisch vor, ist aber sicherlich so vorgesehen?

**Review by JannikUekoetter — APPROVED** [review]
Hey Kai, 
I looked at your PR and found only small things. Otherwise, looks great. Keep up this cool work. I am proud of you!

File: pages/user.js (hunk 1 above)
Comment (kai-franke) [reply pages/user.js]: Thank you for pointing this out. I have taken your suggestion as an opportunity to make this consistent throughout the app.

**Comment by kai-franke** [comment]
> Hey Kai, I looked at your PR and found only small things. Otherwise, looks great. Keep up this cool work. I am proud of you!

Thank you for your warm words, @JannikUekoetter. I am also proud of you!

File: pages/user.js (hunk 2 above)
Comment (kai-franke) [reply pages/user.js:79]: ESLint suggests this notation instead of an apostrophe.

**Pull request merged**